sled-agent/src/services.rs

```diff
@@ -778,26 +794,45 @@ impl ServiceManager {
         // lock.
         _map: &MutexGuard<'_, ZoneMap>,
     ) -> Result<Option<Ledger<OmicronZonesConfigLocal>>, Error> {
-        // First, try to load the current software's zone ledger.  If that
-        // works, we're done.
+        // NOTE: This is a function where we used to access zones by "service
+        // ledgers". This format has since been deprecated, and these files,
+        // if they exist, should not be used.
+        //
+        // We try to clean them up at this spot. Deleting this "removal" code
+        // in the future should be a safe operation; this is a non-load-bearing
+        // cleanup.
+        for path in self.all_service_ledgers().await {
+            match tokio::fs::remove_file(&path).await {
+                Ok(_) => (),
+                Err(ref e) if e.kind() == std::io::ErrorKind::NotFound => (),
+                Err(e) => {
+                    return Err(Error::io(
+                        "Failed to delete old service ledger",
+                        e,
+                    ))
```

Comment:
I'd just `warn!` here and proceed, since we don't actually need it to be cleaned up.

Reply:
Done: a04e9c79f3aefba7f18be982fee09986a1c5d3bf